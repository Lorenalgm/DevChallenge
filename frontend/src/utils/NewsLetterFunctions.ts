import { addDoc, collection, getDocs, query, where } from "firebase/firestore";
import { db } from "@/services/firebase-config";

async function addEmailIntoNewsletter(email: string) {
  try {
    // Verificando se o email ja está cadastrado
    const newsletterRef = collection(db, "newsletter");
    const newsletterSnap = await getDocs(
      query(newsletterRef, where("email", "==", email))
    )
      .then((snap) => {
        const newsletterArray = [];
        snap.docs.forEach((doc) => {
          newsletterArray.push(doc.data());
        });

        return newsletterArray.length === 0;
      })
      .catch((error) => {
        console.log("newsletterSnap error", error);
      });

    if (!newsletterSnap) {
      alert("Parece que você já cadastrou este email.");

      return false;
    }

    const data = {
      email,
      optInDate: Date.now(),
      active: true,
      optOutDate: "",
    };

    await addDoc(newsletterRef, data);

    return true;
  } catch (error) {
    console.log("addEmailIntoNewsletter", error);
    return false;
  }
}

export { addEmailIntoNewsletter };
