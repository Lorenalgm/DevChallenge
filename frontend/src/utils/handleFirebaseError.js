export const AuthErrorHandler = {
    generic:
        'Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.',
    'auth/weak-password':
        'Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.',
    'auth/email-already-in-use': 'O email fornecido ja está em uso.',
    'auth/email-already-exists':
        'O e-mail fornecido já está em uso por outro usuário.',
    'auth/internal-error':
        'Erro interno do servidor. Tente novamente mais tarde.',
    'auth/invalid-email': 'Email inválido.',
    'auth/user-not-found':
        'Não há registro de usuário existente correspondente ao identificador fornecido.',
    'auth/wrong-password': 'Senha digitada não esta correta',
    'resource-exhausted':
        'Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.',
};
