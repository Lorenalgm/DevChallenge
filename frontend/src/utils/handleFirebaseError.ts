export const AuthErrorHandler = (option: string) => {
  switch (option) {
    case "generic":
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
    case "auth/weak-password":
      return "Senha digitada é muito fraca. Tente usar mais de 6 caracteres com números e caracteres especiais.";
    case "auth/email-already-in-use":
      return "O email fornecido ja está em uso.";
    case "auth/email-already-exists":
      return "O e-mail fornecido já está em uso por outro usuário.";
    case "auth/internal-error":
      return "Erro interno do servidor. Tente novamente mais tarde.";
    case "auth/invalid-email":
      return "Email inválido.";
    case "auth/user-not-found":
      return "Não há registro de usuário existente correspondente ao identificador fornecido.";
    case "auth/wrong-password":
      return "Senha digitada não esta correta";
    case "resource-exhausted":
      return "Servidores em manutenção. Se esse erro persistir, contate o administrador do sistema.";
    default:
      return "Houve um erro ao cadastrar o usuario. Tente novamente mais tarde.";
  }
};
