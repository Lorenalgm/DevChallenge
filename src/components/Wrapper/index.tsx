import { Flex, FlexProps } from '@chakra-ui/react'

const Wrapper = ({ children, ...props }: FlexProps) => (
  <Flex
    width="90%"
    maxWidth="container.xl"
    mx="auto"
    flexDirection="column"
    {...props}
  >
    {children}
  </Flex>
)

export default Wrapper
