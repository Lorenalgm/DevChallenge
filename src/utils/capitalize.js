export default function capitalize(word) {
    if (!word) return undefined;

    const firstLetter = word[0];
    const restOfWord = word.slice(1);

    return firstLetter.toUpperCase() + restOfWord;
}
