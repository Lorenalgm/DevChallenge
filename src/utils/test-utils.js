import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { ToastContainer } from 'react-toastify';
import { render } from '@testing-library/react';

import { GlobalStyles } from '../styles/GlobalStyles';

const customRender = (component, { ...renderOptions } = {}) =>
    render(
        <>
            <GlobalStyles />
            {component}
            <ToastContainer />
        </>,
        { wrapper: MemoryRouter, ...renderOptions }
    );

export * from '@testing-library/react';
export { customRender as render };
