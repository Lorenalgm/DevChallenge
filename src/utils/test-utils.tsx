import { render, RenderResult } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import { matchers } from '@emotion/jest'

expect.extend(matchers)

import theme from 'theme'

export const customRender = (children: React.ReactNode): RenderResult =>
  render(
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  )

export * from '@testing-library/react'
export { customRender as render }
