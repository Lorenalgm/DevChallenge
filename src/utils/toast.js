import { toast } from 'react-toastify';

const messages = {
  error(msg) {
    return msg || 'Ocorreu um erro no processamento';
  },
  success(msg) {
    return msg || 'Sucesso';
  },
};

const ToastNotification = {
  notify: (method, msg) => {
    const methodStr = method;
    method = toast[method];
    method(messages[methodStr](msg));
  },
};

export default ToastNotification;
