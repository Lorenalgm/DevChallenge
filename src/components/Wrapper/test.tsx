import { render, screen } from 'utils/test-utils'

import Wrapper from '.'

describe('<Wrapper />', () => {
  it('should render the children component', () => {
    const { container } = render(
      <Wrapper>
        <div data-testid="children-component"></div>
      </Wrapper>
    )

    expect(screen.getByTestId('children-component')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
