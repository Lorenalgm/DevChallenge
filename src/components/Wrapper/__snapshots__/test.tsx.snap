// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Wrapper /> should render the children component 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 90%;
  max-width: var(--chakra-sizes-container-xl);
  -webkit-margin-start: auto;
  margin-inline-start: auto;
  -webkit-margin-end: auto;
  margin-inline-end: auto;
}

<div
  class="emotion-0"
>
  <div
    data-testid="children-component"
  />
</div>
`;
